### PR TITLE
WIP places database

### PR DESCRIPTION
**Pull request by benfrancis**

### CODE REVIEW

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=eb43d0745c47f796f574c27274fb62180879ab68

File: apps/system/js/places.js [hunk 1]
```diff
@@ -1,586 +1,190 @@
...
+      this.db = event.target.result;
+  
+      if(this.db.objectStoreNames.contains(this.SITES_STORE)) {
+        console.log('Sites store already exists');
+        return;
+      }
+  
+      var objectStore = this.db.createObjectStore(this.SITES_STORE,
+        { keyPath: 'id' });
+      objectStore.createIndex('frecency', 'frecency', { unique: false });
```
Comment (albertopq) [inline apps/system/js/places.js]: Should we create a 'pinned' index, in order to get all the pinned sites?

File: apps/system/js/places.js [hunk 2]
```diff
@@ -1,586 +1,190 @@
...
 
-      switch (evt.type) {
+      switch (event.type) {
         case 'applocationchange':
-          this.onLocationChange(app.config.url);
+          this.handleLocationChange(browserWindow.config.url);
           break;
-        case 'apptitlechange':
-          this.onTitleChange(app.config.url, app.title);
+        ase 'apptitlechange':
```
Comment (pochy-ja) [inline apps/system/js/places.js]: I think here is need "c" for the correct case, right?

**Pull request closed without merging**